tests/jni_api.rs

```diff
@@ -228,3 +227,160 @@ pub fn get_object_class_null_arg() {
+    const SIZE: jsize = 16;
+
+    let result = env.new_boolean_array(SIZE);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+
+    let result = env.new_byte_array(SIZE);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+
+    let result = env.new_char_array(SIZE);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+
+    let result = env.new_short_array(SIZE);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+
+    let result = env.new_int_array(SIZE);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+
+    let result = env.new_long_array(SIZE);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+
+    let result = env.new_float_array(SIZE);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+
+    let result = env.new_double_array(SIZE);
+    assert!(result.is_ok());
+    assert!(!result.unwrap().is_null());
+}
+
+// Group test for testing the family of new_PRIMITIVE_array functions with wrong arguments
+#[test]
+pub fn new_primitive_array_wrong() {
+    let env = attach_current_thread();
+    const WRONG_SIZE: jsize = -1;
+
+    let result = env.new_boolean_array(WRONG_SIZE);
+    assert!(result.is_err());
+    assert_error(result, "JNIEnv#new_boolean_array should throw exception");
+    assert_pending_java_exception(&env);
+
+    let result = env.new_byte_array(WRONG_SIZE);
+    assert!(result.is_err());
+    assert_error(result, "JNIEnv#new_byte_array should throw exception");
+    assert_pending_java_exception(&env);
+
+    let result = env.new_char_array(WRONG_SIZE);
+    assert!(result.is_err());
+    assert_error(result, "JNIEnv#new_char_array should throw exception");
+    assert_pending_java_exception(&env);
+
+    let result = env.new_short_array(WRONG_SIZE);
+    assert!(result.is_err());
+    assert_error(result, "JNIEnv#new_short_array should throw exception");
+    assert_pending_java_exception(&env);
+
+    let result = env.new_int_array(WRONG_SIZE);
+    assert!(result.is_err());
+    assert_error(result, "JNIEnv#new_int_array should throw exception");
+    assert_pending_java_exception(&env);
+
+    let result = env.new_long_array(WRONG_SIZE);
+    assert!(result.is_err());
+    assert_error(result, "JNIEnv#new_long_array should throw exception");
+    assert_pending_java_exception(&env);
+
+    let result = env.new_float_array(WRONG_SIZE);
+    assert!(result.is_err());
+    assert_error(result, "JNIEnv#new_float_array should throw exception");
+    assert_pending_java_exception(&env);
+
+    let result = env.new_double_array(WRONG_SIZE);
+    assert!(result.is_err());
+    assert_error(result, "JNIEnv#new_double_array should throw exception");
+    assert_pending_java_exception(&env);
+}
+
+// Helper method that asserts that result is Error and the cause is JavaException.
+fn assert_error(res: Result<jobject, Error>, expect_message: &str) {
```

Comment:
I'd pick a more specific name — not any Err is accepted, but the one corresponding to an exception. Maybe, `assert_exception`, that will also delegate to `assert_pending_exception`?

Reply:
Yes, it's more related to the exception, I agree. Thanks, renamed to assert_exception.